### PR TITLE
Fix `cupy.linalg.qr` to align with NumPy 1.22

### LINKED ISSUE #6198: Support NumPy 1.22
## [Tasks](https://github.com/cupy/cupy/wiki/Actions-Needed-for-Dependency-Update)

- [x] Read Release Notes.
- [x] Update FlexCI schema/matrix to test against the version. #6323
- [x] For NumPy/SciPy, update requirements in `setup.py`.
  - [x] Update the pinned upper bound version (see #4773) #6323
  - [x] Backport. #6354
- [x] Update documentation
  - [x] Update Requirements section in [Installation Guide](https://docs.cupy.dev/en/latest/install.html). The baseline version must be updated in the development branch. #6323
  - [x] Update `docs/requirements.txt` to use the latest versions. #6323
  - [x] Backport. Do not update the baseline version in backport. #6354
- [x] Update CuPy codebase (if needed) and backport.

----

Test result with NumPy 1.22.0rc1: https://ci.preferred.jp/cupy.linux.cuda-head/88054/

Output dtype must be `float32` instead of `float64` (ref #5583):

```
cupy_tests/linalg_tests/test_decomposition.py::TestQRDecomposition_param_1_{mode='raw'}::test_empty_array	
cupy_tests/linalg_tests/test_decomposition.py::TestQRDecomposition_param_1_{mode='raw'}::test_mode	
cupy_tests/linalg_tests/test_decomposition.py::TestQRDecomposition_param_1_{mode='raw'}::test_mode_rank3	
cupy_tests/linalg_tests/test_decomposition.py::TestQRDecomposition_param_1_{mode='raw'}::test_mode_rank4	
```

Add `method` option (`interpolation` deprecated) in `quantile` and `percentile`:

```
cupy_tests/statistics_tests/test_order.py::TestPercentileMonotonic_param_0_{magic_value=-29}::test_percentile_monotonic	
cupy_tests/statistics_tests/test_order.py::TestPercentileMonotonic_param_1_{magic_value=-53}::test_percentile_monotonic	
cupy_tests/statistics_tests/test_order.py::TestPercentileMonotonic_param_2_{magic_value=-207}::test_percentile_monotonic	
cupy_tests/statistics_tests/test_order.py::TestPercentileMonotonic_param_3_{magic_value=-16373}::test_percentile_monotonic	
cupy_tests/statistics_tests/test_order.py::TestPercentileMonotonic_param_4_{magic_value=-99999}::test_percentile_monotonic	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_percentile_bad_q	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_percentile_defaults	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_percentile_keepdims	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_percentile_neg_axis	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_percentile_no_axis	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_percentile_out	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_percentile_out_of_range_q	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_percentile_q_list	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_percentile_scalar_q	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_percentile_tuple_axis	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_percentile_unexpected_interpolation	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_quantile_bad_q	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_quantile_defaults	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_quantile_keepdims	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_quantile_neg_axis	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_quantile_no_axis	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_quantile_out	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_quantile_out_of_range_q	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_quantile_q_list	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_quantile_scalar_q	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_quantile_tuple_axis	
cupy_tests/statistics_tests/test_order.py::TestOrder::test_quantile_unexpected_interpolation	
```

### PR DESCRIPTION
**Pull request by takagi**
Partially fix #6198.

### CODE REVIEW

File: tests/cupy_tests/linalg_tests/test_decomposition.py [hunk 1]
```diff
@@ -127,14 +127,23 @@ def check_mode(self, array, mode, dtype, batched=False):
         a_gpu = cupy.asarray(array, dtype=dtype)
         result_gpu = cupy.linalg.qr(a_gpu, mode=mode)
 
-        if ((not batched)
-                or (numpy.lib.NumpyVersion(numpy.__version__) >= '1.22.0')):
+        if numpy.lib.NumpyVersion(numpy.__version__) >= '1.22.0a1':
```
Comment (takagi) [inline tests/cupy_tests/linalg_tests/test_decomposition.py:130]: 1.22.0rc1 < 1.22.0 in NumpyVersion so we have to use a pre-released version string to test against release candidates.

File: tests/cupy_tests/linalg_tests/test_decomposition.py [hunk 2]
```diff
@@ -127,14 +127,23 @@ def check_mode(self, array, mode, dtype, batched=False):
         a_gpu = cupy.asarray(array, dtype=dtype)
         result_gpu = cupy.linalg.qr(a_gpu, mode=mode)
 
-        if ((not batched)
-                or (numpy.lib.NumpyVersion(numpy.__version__) >= '1.22.0')):
+        if numpy.lib.NumpyVersion(numpy.__version__) >= '1.22.0rc1':
```
Comment (takagi) [inline tests/cupy_tests/linalg_tests/test_decomposition.py:130]: 1.22.0rc1 < 1.22.0 in NumpyVersion so we have to use a pre-released version string to test against release candidates.

**Review by toslunar — APPROVED** [review]
LGTM

**Comment by toslunar** [comment]
`cupy_tests/linalg_tests/test_decomposition.py` passed. https://ci.preferred.jp/cupy.linux.cuda-head/89422/#L3965

/test mini

**Comment by leofang** [comment]
Looks like there are still a few tests to be fixed:
https://ci.preferred.jp/cupy.linux.cuda11x-cuda-python/89785/

**Pull request merged**